### PR TITLE
Zendesk (Source) initial commit

### PR DESCRIPTION
**Pull request by mohan-h**
Zendesk (Source) initial commit

### CODE REVIEW

File: help/sources/tutorials/ui/create/customer-success/zendesk.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+---
+keywords: Experience Platform;Zendesk;sources;connectors;source connectors;sources sdk;sdk;SDK
+title: Create a Zendesk source connection in the UI
+description: Learn how to create a Zendesk source connection using the Adobe Experience Platform UI.
+hide: true
+hidefromtoc: true
+exl-id:
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/customer-success/zendesk.md]: We can remove "exl-id" here as it is auto-generated upon publishing

File: help/sources/tutorials/ui/create/customer-success/zendesk.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+---
+# Create a Zendesk source connection in the UI
+
+This tutorial provides steps for creating a Zendesk source connector using the Platform user interface.
+
+## Overview
+
+[Zendesk](https://www.zendesk.com) is a popular customer service solution and sales tool.
+
+This [!DNL Adobe Experience Platform] [!DNL sources connector](https://experienceleague.adobe.com/docs/experience-platform/sources/home.html?lang=en) leverages the [!DNL Zendesk Search API > Export Search Results](https://developer.zendesk.com/api-reference/ticketing/ticket-management/search/#export-search-results) that returns users information into Experience Platform from Zendesk for further processing.
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/customer-success/zendesk.md]: We can remove the [!DNL] tags here for Adobe Experience Platform and source connector. Also let's reword "sources" to just sources. The [!DNL] tag in "Zendesk Search API" can also be removed.

File: help/sources/tutorials/ui/create/customer-success/zendesk.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+Zendesk uses bearer tokens as an authentication mechanism to communicate with the Zendesk API.
+
+## Prerequisites
+
+The following items are required before you start configuring the extension:
+
+* You need to have a Zendesk Support account. 
+    * Go to the Zendesk [register](https://www.zendesk.com/register/) page to register and create a Zendesk account, if you do not have one already.
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/customer-success/zendesk.md]: Since there is only one prerequisite, we can restructure this to just one short paragraph. Something like: "Before you can configure a Zendesk source connection, you must first ensure that you have Zendesk Support account. If you do not have a Zendesk Support account, go to the [Zendesk register](https://www.zendesk.com/register/) to register and create your Zendesk Support account before attempting to do this tutorial.

File: help/sources/tutorials/ui/create/customer-success/zendesk.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+### Gather required credentials
+
+In order to connect Zendesk to Platform, you must provide values for the following connection properties:
+
+| Credential | Description | Example |
+| --- | --- | --- |
+| subdomain | Unique domain specific to your account created during the registration process. | *xxxxx.zendesk.com*|
+| API token | Zendesk API token.<br></br>Navigate to the Zendesk website and access **[!UICONTROL Settings]** > **[!UICONTROL Apps and Integrations]** > **[!UICONTROL Zendesk API]** page. Then Navigate to the **[!UICONTROL API Tokens]** section. |*0lZnClEvkJSTQ7olGLl7PMhVq99gu26GTbJtf*|
+
+![Zendesk domain](../../../../images/tutorials/create/zendesk/zendesk-domain.png?lang=en)
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/customer-success/zendesk.md]: We can probably just skip this domain screenshot and just add a link to "Zendesk website" in the credential tables

File: help/sources/tutorials/ui/create/customer-success/zendesk.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+## Additional resources
+
+### Validation {#validation}
+
+To validate that you have correctly set up the source; and Zendesk profiles are being ingested, follow the steps below:
+
+1. In the Platform UI, select **[!UICONTROL Datasets]** from the left navigation bar to access the [!UICONTROL Datasets] workspace. The [!UICONTROL Dataset Activity] screen displays the details of executions.
+![Activity page](../../../../images/tutorials/create/zendesk/dataset-activity.png?lang=en)
+![Dataflow page](../../../../images/tutorials/create/zendesk/dataflow-monitoring.png?lang=en)
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/customer-success/zendesk.md]: Let's add a sentence between these two screenshots, describing the step that transitions between them. Something along the lines of: "Next, select the dataflow run ID of the dataflow that you want to view, to see specific details about that dataflow run."

File: help/sources/tutorials/ui/create/customer-success/zendesk.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+1. Next, select the **[!UICONTROL Data governance]** tab, you will see the schema details.
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/customer-success/zendesk.md]: We can rephrase this to something like: "the **[!UICONTROL Data governance]** tab to see the schema details."

File: help/sources/tutorials/ui/create/customer-success/zendesk.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+![Zendesk schema](../../../../images/tutorials/create/zendesk/dataset-governance.png?lang=en)
+
+1. Finally, select the **[!UICONTROL Preview dataset]** button to display the data that was ingested.
+![Zendesk dataset](../../../../images/tutorials/create/zendesk/preview-dataset.png?lang=en)
+
+### Zendesk schema
+
+* A list of supported mappings that are to be set up for the [!DNL Zendesk Search API > Export Search Results](https://developer.zendesk.com/api-reference/ticketing/ticket-management/search/#export-search-results) is given below.
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/customer-success/zendesk.md]: Can we rephrase this to something like: "The table below outlines a list of supported mappings that must be set up for the [!DNL Zendesk Search API > Export Search Results](https://developer.zendesk.com/api-reference/ticketing/ticket-management/search/#export-search-results)."

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 8]
```diff
@@ -0,0 +1,1159 @@
+---
+keywords: Experience Platform;home;popular topics;sources;connectors;source connectors;sources sdk;sdk;SDK
+solution: Experience Platform
+title: Documentation self-service template
+topic-legacy: tutorial
+description: Learn how to connect Adobe Experience Platform to Zendesk using the Flow Service API.
+hide: true
+hidefromtoc: true
+exl-id:
```
Comment (carlojavier) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: We can remove "exl-id" here as it is auto-generated.

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 9]
```diff
@@ -0,0 +1,1159 @@
...
+hidefromtoc: true
+exl-id: 
+---
+# Create a Zendesk connection using the Flow Service API
+
+## Overview
+
+[Zendesk](https://www.zendesk.com) is a popular customer service solution and sales tool.
+
+This [!DNL Adobe Experience Platform] [!DNL sources connector](https://experienceleague.adobe.com/docs/experience-platform/sources/home.html?lang=en) leverages the [!DNL Zendesk Search API > Export Search Results](https://developer.zendesk.com/api-reference/ticketing/ticket-management/search/#export-search-results) that returns users information into Experience Platform from Zendesk for further processing.
```
Comment (carlojavier) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: We can remove the three DNL tags in this paragraph and let's reword "sources" to just "source"

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 10]
```diff
@@ -0,0 +1,1159 @@
...
+### Gather required credentials
+
+In order to connect Zendesk to Platform, you must provide values for the following connection properties:
+
+| Credential | Description | Example |
+| --- | --- | --- |
+| subdomain | Unique domain specific to your account created during the registration process. | *xxxxx.zendesk.com*|
+| API token | Zendesk API token.<br></br>Navigate to the Zendesk website and access **[!UICONTROL Settings]** > **[!UICONTROL Apps and Integrations]** > **[!UICONTROL Zendesk API]** page. Then Navigate to the **[!UICONTROL API Tokens]** section. |*0lZnClEvkJSTQ7olGLl7PMhVq99gu26GTbJtf*|
+
+![Zendesk domain](../../../../images/tutorials/create/zendesk/zendesk-domain.png?lang=en)
```
Comment (carlojavier) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: Like the UI doc, we can remove this domain screenshot and just add a link to "Zendesk website" in the screenshot

**Review by carlojavier — CHANGES_REQUESTED** [review]
Hey @mohan-h thanks for this! Added some comments, just some small changes. Thanks!

File: help/sources/tutorials/ui/create/customer-success/zendesk.md [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+
+Before you start configuring the extension you need to have a Zendesk Support account. If you do not have one already go to the Zendesk [register](https://www.zendesk.com/register/) page to register and create your Zendesk account.
+
+### Gather required credentials
+
+In order to connect Zendesk to Platform, you must provide values for the following connection properties:
+
+| Credential | Description | Example |
+| --- | --- | --- |
+| subdomain | Unique domain specific to your account created during the registration process. <br/><br/> Refer to the [Zendesk documentation](https://support.zendesk.com/hc/en-us/articles/4409381383578-Where-can-I-find-my-Zendesk-subdomain-) if you require any guidance. | *xxxxx.zendesk.com*|
```
Comment (vivetiwa) [inline help/sources/tutorials/ui/create/customer-success/zendesk.md]: It should be inform of url, https should be added to host with the subdomain.
https://support.zendesk.com/hc/en-us/articles/4409381383578-Where-can-I-find-my-Zendesk-subdomain-#:~:text=https%3A//yoursubdomain.zendesk.com

File: help/sources/tutorials/ui/create/customer-success/zendesk.md [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+
+### New account
+
+If you are creating a new account, select **[!UICONTROL New account]**, and then provide a name, an optional description, and your credentials. When finished, select **[!UICONTROL Connect to source]** and then allow some time for the new connection to establish.
+
+![new](../../../../images/tutorials/create/zendesk/authentication-new-account.png)
+
+## Next steps
+
+By following this tutorial, you have established a connection to your *Zendesk* account. You can now continue on to the next tutorial and [configure a dataflow to bring data into Platform](https://experienceleague.adobe.com/docs/experience-platform/sources/ui-tutorials/dataflow/crm.html).
```
Comment (vivetiwa) [inline help/sources/tutorials/ui/create/customer-success/zendesk.md]: @carlojavier for rest based connector preview screen input will be somewhat different. There would be no table or folder selection as it is in CRM and cloud based sources.

In rest user either needs to provide some input parameters or if input params are not there  then he will see the preview of the data.

Should be there preview screen in documentation as we have added for mailchimp campaign similar to that section defining the preview of zendesk ? - 

https://experienceleague.adobe.com/docs/experience-platform/sources/ui-tutorials/create/marketing-automation/mailchimp-campaigns.html?lang=en#:~:text=Once%20your%20source%20is%20authenticated

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 13]
```diff
@@ -0,0 +1,1152 @@
...
+    -d '{
+        "version": 0,
+        "xdmSchema": "https://ns.adobe.com/{TENANT_ID}/schemas/995dabbea86d58e346ff91bd8aa741a9f36f29b1019138d4",
+        "xdmVersion": "1.0",
+        "id": null,
+        "mappings": [
+    "mappings": [
+        {
+            "sourceType": "ATTRIBUTE",
+            "source": "results.id",
```
Comment (vivetiwa) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: `results` should be replace with `result` as it would be single object not an array.

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 14]
```diff
@@ -0,0 +1,1152 @@
...
+    -H 'Authorization: Bearer {ACCESS_TOKEN}' \
+    -H 'x-api-key: {API_KEY}' \
+    -H 'x-gw-ims-org-id: {IMS_ORG}' \
+    -H 'x-sandbox-name: {SANDBOX_NAME}' \
+    -H 'Content-Type: application/json' \
+    -d '{
+        "name": "Zendesk base connection",
+        "description": "Zendesk base connection to authenticate to Platform",
+        "connectionSpec": {
+            "id": "2526ee57-0af4-46ea-a477-720a53108145",
```
Comment (vivetiwa) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: connectionSpec id should be - `0a27232b-2c6e-4396-b8c6-c9fc24e37ba4`

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 15]
```diff
@@ -0,0 +1,1152 @@
...
+            "version": "1.0"
+        },
+        "auth": {
+            "specName": "OAuth generic-rest-connector",
```
Comment (vivetiwa) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: specName should be - `OAuth2 Refresh Code`

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 16]
```diff
@@ -0,0 +1,1152 @@
...
+            "params": {
+                "host":{ZENDESK_SUBDOMAIN}
```
Comment (vivetiwa) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: missing double quotes - `"host": "{ZENDESK_SUBDOMAIN}"`

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 17]
```diff
@@ -0,0 +1,1152 @@
...
+    }'
+```
+
+| Property | Description |
+| --- | --- |
+| `name` | The name of your base connection. Ensure that the name of your base connection is descriptive as you can use this to look up information on your base connection. |
+| `description` | An optional value that you can include to provide more information on your base connection. |
+| `connectionSpec.id` | The connection specification ID of your source. This ID can be retrieved after your source is registered and approved through the [!DNL Flow Service] API. |
+| `auth.specName` | The authentication type that you are using to authenticate your source to Platform. |
+| `auth.params.` | Contains the credentials required to authenticate your source. |
```
Comment (vivetiwa) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: `auth.params.` params missing.
 Please refer : https://experienceleague.adobe.com/docs/experience-platform/sources/api-tutorials/create/marketing-automation/mailchimp-campaign.html?lang=en#:~:text=authenticate%20your%20source%20to%20Platform.-,auth.params.host,-The%20root%20URL%20used%20to

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 18]
```diff
@@ -0,0 +1,1152 @@
...
+
+A successful response returns the structure of the queried file. In the example below within the ``data[]`` payload only a single record is shown, however there could be multiple records.
+
+```json
+{
+    "format": "hierarchical",
+    "schema": {
+        "type": "object",
+        "properties": {
+            "results": {
```
Comment (vivetiwa) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: `results` should be `result`

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 19]
```diff
@@ -0,0 +1,1152 @@
...
+            }
+        }
+    },
+    "data": [
+        {
+            "meta": {
+                "has_more": true,
+                "after_cursor": "eyJmaWVsZCI6ImNyZWF0ZWRfYXQiLCJkZXNjIjp0cnVlLCJ0aWVCcmVha0ZpZWxkIjoiaWQiLCJ0aWVCcmVha0Rlc2MiOmZhbHNlLCJzb3J0VmFsdWVzIjpbMTY1MjE3MjA4NTAwMCw2MDMwODg3MTM5NzI5XSwiZXhwb3J0ZWRUaHVzRmFyIjoxMDAsInNlc3Npb25TdGFydCI6MTY1MjQzNDE4MzkwOCwiY3JlYXRlZEF0IjoxNjUyNDM0MTgzOTI2LCJzYWx0ZWRSZXF1ZXN0SGFzaCI6LTczMzQxODQ0NSwic2FsdGVkQ3Vyc29ySGFzaCI6MjgxNzM0NzU2fQ=="
+            },
+            "links": {
```
Comment (vivetiwa) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: links, meta should not be visible as part of preview. Could be removed from preview.

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 20]
```diff
@@ -0,0 +1,1152 @@
...
+                "next": "https://d3v-dxinfosys.zendesk.com/api/v2/search/export.json?filter%5Btype%5D=user&page%5Bafter%5D=eyJmaWVsZCI6ImNyZWF0ZWRfYXQiLCJkZXNjIjp0cnVlLCJ0aWVCcmVha0ZpZWxkIjoiaWQiLCJ0aWVCcmVha0Rlc2MiOmZhbHNlLCJzb3J0VmFsdWVzIjpbMTY1MjE3MjA4NTAwMCw2MDMwODg3MTM5NzI5XSwiZXhwb3J0ZWRUaHVzRmFyIjoxMDAsInNlc3Npb25TdGFydCI6MTY1MjQzNDE4MzkwOCwiY3JlYXRlZEF0IjoxNjUyNDM0MTgzOTI2LCJzYWx0ZWRSZXF1ZXN0SGFzaCI6LTczMzQxODQ0NSwic2FsdGVkQ3Vyc29ySGFzaCI6MjgxNzM0NzU2fQ%3D%3D&query=type%3Auser"
+            },
+            "results": {
```
Comment (vivetiwa) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: `results` to `result`

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 21]
```diff
@@ -0,0 +1,1152 @@
...
+    -H 'x-api-key: {API_KEY}' \
+    -H 'x-gw-ims-org-id: {IMS_ORG}' \
+    -H 'x-sandbox-name: {SANDBOX_NAME}' \
+    -H 'Content-Type: application/json' \
+    -d '{
+        "name": "Zendesk Source Connection",
+        "description": "Zendesk Source Connection",
+        "baseConnectionId": "70383d02-2777-4be7-a309-9dd6eea1b46d",
+        "connectionSpec": {
+            "id": "6360f136-5980-4111-8bdf-15d29eab3b5a",
```
Comment (vivetiwa) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: connectionSpec id should be - `0a27232b-2c6e-4396-b8c6-c9fc24e37ba4`

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 22]
```diff
@@ -0,0 +1,1152 @@
...
+    -H 'x-api-key: {API_KEY}' \
+    -H 'x-gw-ims-org-id: {IMS_ORG}' \
+    -H 'x-sandbox-name: {SANDBOX_NAME}' \
+    -H 'Content-Type: application/json' \
+    -d '{
+        "version": 0,
+        "xdmSchema": "https://ns.adobe.com/{TENANT_ID}/schemas/995dabbea86d58e346ff91bd8aa741a9f36f29b1019138d4",
+        "xdmVersion": "1.0",
+        "id": null,
+        "mappings": [
```
Comment (vivetiwa) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: duplicate "mappings" attributes

File: help/sources/tutorials/api/create/customer-success/zendesk.md [hunk 23]
```diff
@@ -0,0 +1,1152 @@
...
+    -H 'Authorization: Bearer {ACCESS_TOKEN}' \
+    -H 'x-api-key: {API_KEY}' \
+    -H 'x-gw-ims-org-id: {IMS_ORG}' \
+    -H 'x-sandbox-name: {SANDBOX_NAME}' \
+    -H 'Content-Type: application/json' \
+    -d '{
+        "version": 0,
+        "xdmSchema": "https://ns.adobe.com/{TENANT_ID}/schemas/995dabbea86d58e346ff91bd8aa741a9f36f29b1019138d4",
+        "xdmVersion": "1.0",
+        "id": null,
```
Comment (vivetiwa) [inline help/sources/tutorials/api/create/customer-success/zendesk.md]: `"id": null` could be removed

File: help/sources/tutorials/ui/create/customer-success/zendesk.md (hunk 12 above)
Comment (carlojavier) [reply help/sources/tutorials/ui/create/customer-success/zendesk.md]: @vivetiwa good call, thanks for heads up on this. We'll definitely update this page to be similar to the mailchimp doc structure

File: help/sources/tutorials/api/create/customer-success/zendesk.md (hunk 13 above)
Comment (carlojavier) [reply help/sources/tutorials/api/create/customer-success/zendesk.md]: Does this apply to all values for `source` in this section or just `results.id`?

File: help/sources/tutorials/api/create/customer-success/zendesk.md (hunk 20 above)
Comment (mohan-h) [reply help/sources/tutorials/api/create/customer-success/zendesk.md]: Checked the curl response, it is `results`. I'll share the API request, response to you on slack.

File: help/sources/tutorials/api/create/customer-success/zendesk.md (hunk 13 above)
Comment (mohan-h) [reply help/sources/tutorials/api/create/customer-success/zendesk.md]: @vivetiwa This was the exact response from the curl request

File: help/sources/tutorials/api/create/customer-success/zendesk.md (hunk 16 above)
Comment (mohan-h) [reply help/sources/tutorials/api/create/customer-success/zendesk.md]: Good spot.

File: help/sources/tutorials/api/create/customer-success/zendesk.md (hunk 22 above)
Comment (mohan-h) [reply help/sources/tutorials/api/create/customer-success/zendesk.md]: Good spot.

File: help/sources/tutorials/api/create/customer-success/zendesk.md (hunk 17 above)
Comment (mohan-h) [reply help/sources/tutorials/api/create/customer-success/zendesk.md]: Thanks @vivetiwa, added.
@carlojavier the template doesn't provide guidance on this. Suggest to add guidance.

File: help/sources/tutorials/api/create/customer-success/zendesk.md (hunk 18 above)
Comment (mohan-h) [reply help/sources/tutorials/api/create/customer-success/zendesk.md]: curl response has the attribute as results.

File: help/sources/tutorials/ui/create/customer-success/zendesk.md (hunk 12 above)
Comment (mohan-h) [reply help/sources/tutorials/ui/create/customer-success/zendesk.md]: @carlojavier sorry didn't understand need guidance.

**Comment by mohan-h** [comment]
The documentation for the connector has been made live at:
https://experienceleague.adobe.com/docs/experience-platform/sources/connectors/customer-success/zendesk.html?lang=en
